src/CodeFragment.ts

```diff
@@ -1,17 +1,21 @@
-import {CodeToken, isCodeToken, stringToToken} from '@components/CodeToken';
+import {
+  CodeMetrics,
+  isCodeMetrics,
+  measureString,
+} from '@components/CodeMetrics';
 
 export interface CodeFragment {
-  before: CodeToken;
-  after: CodeToken;
+  before: CodeMetrics;
+  after: CodeMetrics;
 }
 
 export type PossibleCodeFragment =
   | CodeFragment
-  | CodeToken
+  | CodeMetrics
   | {before: string; after: string}
   | string;
 
-export function tokenToFragment(value: CodeToken): CodeFragment {
+export function tokenToFragment(value: CodeMetrics): CodeFragment {
```

Comment:
this should probably be `metricsToFragment` as well

Reply:
Done